### DISCUSSION TITLE
Grouping Brews on the User Page

### DISCUSSION THREAD

**Discussion by G-Ambatte** [naturalcrit/homebrewery#1471] [Ideas]
Capturing discussion/thinking from the Gitter chat today for future useL

---

In my thinking, `group` is a special tag. One brew can have multiple` group` tags, but the filter term 'group:A' will only find brews tagged with 'group:A'.
Custom user tags can use [A-Za-z0-9], all symbols reserved for future use, so a tag containing a colon (e.g. `group:erics`) can only be applied through the UI.

The structure itself is still flat, so in this thought experiment, ALL brews are still visible at /user/erics (to the user, and if published, the public) but are automatically filtered to just the `group:fridaygame` at /user/erics/fridaygame.
It's not impossible to imagine a `publish_in_group:true` tag that makes it visible to the public (like PUBLISHED) but only when viewed via the group. A better version being `publish_in_group:fridaygame`, to specify which group to publish it in, rather than just publishing it in all groups it's attached to.

---

**Comment by jeddai** [comment]
What do we think of this regex pattern? 
```regex
^(?:(?:group|meta|system|type):)?[A-Za-z0-9][A-Za-z0-9 /.:-]+$
// group:test-tag = true
// tag            = true
// test:tag       = false
```

**Comment by ericscheid** [comment]
The experience that usually comes to mind when discussing folders (aka "groups") is a thing within which can be found more things (brews, or even more folders) .. but folders also serve the purpose of _unfinding_ things, aka _hiding them from view_. The common use case being “oh no, I have 86 brews on my user page, it’s a jumbled mess”.

Sure, one could search by tags, using tags like folder names .. if you remember the name of the tag .. or by finding that particular folder-tag-name in a big list of tags. Both problems in their own right if you have a lot of tags in your account (which you might well have if you have a lot of brews).

But it would be so much easier if the huge unfiltered mess could be presented as a much smaller collection of buckets.

So .. file that under “just what problem are we trying solve with folders?”
